### DISCUSSION TITLE
Passing property to modal

### DISCUSSION THREAD

**Discussion by gldtn** [wireui/wireui#243] [Q&A]
I'm trying to create an delete/edit button on my table:

**Button:**
```
<x-button.circle flat red icon="trash" value="{{ $contact->id }}" onclick="$openModal('deleteUserModal')" />
```

**Modal:**
```
<x-modal.card title="Delete contact" blur wire:model.defer="deleteUserModal">

    <div class="grid grid-cols-1 sm:grid-cols-2 gap-4">

        Delete contact {{ $contact->id }}

    </div>

    <x-slot name="footer">

        <div class="flex justify-end gap-x-4">

            <div class="flex">

                <x-button flat label="Cancel" x-on:click="close" />

                <x-button negative label="Delete" wire:click="deleteContact({{ $contact->id }})" />

            </div>

        </div>

    </x-slot>

</x-modal.card>
```

But I can not get $contact->id to work, can someone guide me to the right direction here? thanks!

**Comment by jichiduo** [comment]
I suggest you may use livewire way 

Button:
```php
<x-button.circle flat red icon="trash" wire.click="openDeleteUserModal({{$contact->id}})" />
```
Modal:
```php
<x-modal.card title="Delete contact" blur wire:model.defer="deleteUserModal">

    <div class="grid grid-cols-1 sm:grid-cols-2 gap-4">

        Delete contact {{ $selectedID }}

    </div>

    <x-slot name="footer">

        <div class="flex justify-end gap-x-4">

            <div class="flex">

                <x-button flat label="Cancel" x-on:click="close" />

                <x-button negative label="Delete" wire:click="deleteContact" />

            </div>

        </div>

    </x-slot>

</x-modal.card>
```

Component:
```php
namespace ...
use...

class yourClass extends Component
{
    public selectedID;
    ...
    
    public function openDeleteUserModal($id)
    {
         $this->selectedID = $id;
         $this->deleteUserModal = true; 
    }
    public function deleteContact()
    {
        //delete code goes here,below just an example
       Contact::destroy($this->selectedID);
       //cleaning 
       $this->resetValidation();
       $this->reset('selectedID');
       $this->deleteUserModal = true; 
    }
    ...
}
```

Hope can give you some idea

**Reply by gldtn** [reply]
I decided to dive a bit deeper into livewire and I'm on my last screencast video on how to build a datatable. Learning a lot from it.. I was having a hell of a hard time because I didn't quite understand how everything worked :p

Thanks for replying and the examples, always fun to see other coders approach to things.

**Comment by felipemartinspro** [comment]
Hello. I'm developing something similar and I'm not able to reset the validation of the fields inside the modal. What did you do in this function "$this->resetValidation();"
Another thing, after you delete the contact, do you close the modal automatically? If so, what food did you use to do this?
Thank you very much.

**Comment by judgej** [comment]
Thanks, this is helpful. So the `wire:model.defer="deleteUserModal"` refers to a boolean in the component that determines whether the modal is displayed or not? Is this *anywhere* in the documentation? I just looked at the dialogue and modal pages and scratched my head. Feels like a big chunk of "how to make this work" is missing, or I'm just not seeing it.
